internal/discovery/types.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2023 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package discovery
+
+import (
+	"fmt"
+)
+
+// GVKMaps is a struct that holds:
+// * discovery client to fetch objects
+// * Group to Version map
+// * Group to Kind map
+// * Version to Kind map
+// * Kind to Version map
+// * Heartbeat to track the last fetch
+type GVKMaps struct {
```

Comment:
I've kept the wrapper to easily distinguish between instantiation and initialization of the object, since `len` is not defined on `map[V]T` if `T` is a composite data type (we are doing `obj.(GVKMaps) == nil` instead of `len(obj.(map[string]map[string][]string)) == 0` because the latter won't work.